### ISSUE TITLE
Remove multiple withOnyx nestings

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#28824]

**Comment by tgolen** [comment]
I'm doing this internally, so no need for bugzero or C+ yet. I'll unassign you both for now.

**Comment by tgolen** [comment]
I opened the PR for this yesterday and I will continue testing today with a large account to ensure there are no performance problems.

**Comment by tgolen** [comment]
My testing has gone well so I don't forsee there being any performance impact with this change. I'll keep bumping the PR for review.

**Comment by tgolen** [comment]
The Onyx change is currently being rolled out in a separate PR. Once it's rolled out, I will get https://github.com/Expensify/App/pull/28866 ready for review.

**Comment by tgolen** [comment]
It looks like https://github.com/Expensify/App/pull/28894 has been deployed to staging, so I can get back on this again.

**Comment by tgolen** [comment]
That was a mistake. https://github.com/Expensify/App/pull/29169 is the PR with the version bump that this is waiting on.

**Comment by tgolen** [comment]
I'm going to drop this to weekly since it's on HOLD and it's not super critical.

**Comment by tgolen** [comment]
This is still waiting for Onyx to be updated. I had to submit a new fix for some cases where components where infinitely re-rendering: https://github.com/Expensify/react-native-onyx/pull/407

**Comment by tgolen** [comment]
This is still waiting for Onyx to be updated. Janic is looking at the last remaining issues related to the update. More context in [this slack thread](https://expensify.slack.com/archives/C035J5C9FAP/p1698409973062609).

**Comment by tgolen** [comment]
I worked on a new PR today to fix an issue with inifite re-rendering: https://github.com/Expensify/react-native-onyx/pull/412

**Comment by tgolen** [comment]
This is still on HOLD while Onyx gets updated

**Comment by roryabraham** [comment]
Looks like the dependent PR was merged, so I took this off HOLD

**Comment by tgolen** [comment]
Oh, nice! Thanks for taking it off hold. I'll get this one going again.

**Comment by tgolen** [comment]
PR is up and ready for review: https://github.com/Expensify/App/pull/28866

**Comment by laurenreidexpensify** [comment]
Payment Summary:

C+ Review - @situchan $500 offer sent in Upwork - https://www.upwork.com/en-gb/jobs/~01376316f87bc09708

**Comment by laurenreidexpensify** [comment]
Payment Summary:

C+ Review - @situchan $500 paid in Upwork
